### PR TITLE
Update .trivyignore to resolve security alerts

### PR DESCRIPTION
**Pull request by ben-taussig-solo**
# Description
 - Updates `.trivyignore` to ignore [CVE-2022-41721](https://nvd.nist.gov/vuln/detail/CVE-2022-41721)
 - This CVE does not affect the gateway-proxy image, therefore we can ignore it
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/7757

### CODE REVIEW

**Review by ashishb-solo — APPROVED** [review]

**Review by sam-heilbron — CHANGES_REQUESTED** [review]
Can you include steps to verify these changes? I agree that adding this CVE to the trivyignore will skip it, but I found when running locally that other vulnerabilities were identified. I was using trivy version v0.30 (matching CI)

File: changelog/v1.14.0-beta10/security-update.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+changelog:
+  - type: NON_USER_FACING
+    description: >
+      Create .trivyignore file and ignore CVE-2022-41721, a vulnerability in golang.org/x/net.
```
Comment (sam-heilbron) [inline changelog/v1.14.0-beta10/security-update.yaml:4]: Love the description here!

**Comment by ben-taussig-solo** [comment]
> Can you include steps to verify these changes? I agree that adding this CVE to the trivyignore will skip it, but I found when running locally that other vulnerabilities were identified. I was using trivy version v0.30 (matching CI)

The following script validates OSS releases -- it scans the problematic versions, then checks out and builds the fix branches ([v1.11.x fix branch](https://github.com/solo-io/gloo/pull/7820), [v1.10.x fix branch](https://github.com/solo-io/gloo/pull/7819)), then scans the built images. After running this, you can look through `_output/scans` to confirm that the fix branches resolve the issue. There are still issues with some LTS branches in enterprise -- I propose we move ahead with the OSS fixes (I.E, merge this PR are the ones linked above) and then wrap up the enterprise fixes after that.

```sh
# iterate over all affected versions
for oss_version in v1.10.43 v1.11.50 v1.12.42; do
    # create a new directory for each version
    TAGGED_VERSION=$oss_version make scan-version
done

# iterate over fix branches. note that v1.12 does not require a fix branch, as the trivyignore updates allow the scan to go through
for fixed_oss_branch in security-update-yaml-1-10 security-update-yaml-1-11; do
    export VERSION=$fixed_oss_branch
    # checkout and build the version with the fix
    git checkout $fixed_oss_branch
    make docker

    # come back to the security-update-net branch
    git checkout security-update-net

    # scan the fixed version
    make scan-version
done
```

File: Makefile [hunk 2]
```diff
@@ -844,6 +844,13 @@ scan-version: ## Scan all Gloo images with the tag matching {VERSION} env variab
 		-t $(VERSION)\
 		--images gloo,gloo-envoy-wrapper,discovery,ingress,sds,certgen,access-logger,kubectl
 
+.PHONY: scan-enterprise-version
```
Comment (sam-heilbron) [inline Makefile:847]: Happy to have this here if you think it's valuable, but it feels like this leaks implementation details of our enterprise project. There is a make target in the enterprise repo https://github.com/solo-io/solo-projects/blob/master/Makefile#L1781 which you can use to scan those images if you'd like

Comment (ben-taussig-solo) [reply Makefile:847]: gotcha -- I was not aware of the enterprise make target, I will remove this

Comment (ben-taussig-solo) [reply Makefile]: Removed here: https://github.com/solo-io/gloo/pull/7815/commits/f9f9449ac864bceb08bd460290a776676396c52b

**Comment by Bslabe123** [comment]
> > Can you include steps to verify these changes? I agree that adding this CVE to the trivyignore will skip it, but I found when running locally that other vulnerabilities were identified. I was using trivy version v0.30 (matching CI)
> 
> The following script validates OSS releases -- it scans the problematic versions, then checks out and builds the fix branches ([v1.11.x fix branch](https://github.com/solo-io/gloo/pull/7820), [v1.10.x fix branch](https://github.com/solo-io/gloo/pull/7819)), then scans the built images. After running this, you can look through `_output/scans` to confirm that the fix branches resolve the issue. There are still issues with some LTS branches in enterprise -- I propose we move ahead with the OSS fixes (I.E, merge this PR are the ones linked above) and then wrap up the enterprise fixes after that.
> 
> ```shell
> # iterate over all affected versions
> for oss_version in v1.10.43 v1.11.50 v1.12.42; do
>     # create a new directory for each version
>     TAGGED_VERSION=$oss_version make scan-version
> done
> 
> # iterate over fix branches. note that v1.12 does not require a fix branch, as the trivyignore updates allow the scan to go through
> for fixed_oss_branch in security-update-yaml-1-10 security-update-yaml-1-11; do
>     export VERSION=$fixed_oss_branch
>     # checkout and build the version with the fix
>     git checkout $fixed_oss_branch
>     make docker
> 
>     # come back to the security-update-net branch
>     git checkout security-update-net
> 
>     # scan the fixed version
>     make scan-version
> done
> ```

Just verified this script I'm happy to sign off if sam doesnt have any other concerns!

File: .trivyignore [hunk 3]
```diff
@@ -18,4 +18,9 @@ CVE-2022-23524
 # https://github.com/helm/helm/security/advisories/GHSA-53c4-hhmh-vw5q
 CVE-2022-23525
 # https://github.com/helm/helm/security/advisories/GHSA-67fx-wx78-jx33
-CVE-2022-23526
\ No newline at end of file
+CVE-2022-23526
+
+# https://nvd.nist.gov/vuln/detail/CVE-2022-41721
+# Ignore this vulnerability; it does not affect the gateway-proxy image.
+# No handlers exposed by the control plane fall victim to this attack
```
Comment (nfuden) [inline .trivyignore:25]: nit: can we add the info that its due to no usage of the maxbyteshandler?

**Review by sam-heilbron — APPROVED** [review]

File: .trivyignore (hunk 3 above)
Comment (ben-taussig-solo) [reply .trivyignore:25]: resolved here: 442448cbfbbc0a52e538fc96fa69ca95311a6874

**Review by sam-heilbron — APPROVED** [review]

**Review by Bslabe123 — DISMISSED** [review]

**Review by sam-heilbron — APPROVED** [review]

**Review by Bslabe123 — APPROVED** [review]

**Pull request merged**